### PR TITLE
typings fixes | need to pass object as value

### PR DESCRIPTION
**Pull request by dxtr026**

### CODE REVIEW

**Comment by dxtr026** [comment]
@azizhk @ritz078  need to discuss this.

File: src/components/Calendar.tsx [hunk 1]
```diff
@@ -20,11 +20,6 @@ import Button from "./Button";
 import { isSameDay, endOfDay, startOfDay } from "date-fns";
 
 class Calendar extends React.PureComponent<CalendarProps, CalendarState> {
-  static defaultProps: Partial<CalendarProps> = {
```
Comment (azizhk) [inline src/components/Calendar.tsx]: I don't get the rationale behind this.
The PR title probable seems for Controls.ts
Please elaborate.

File: src/components/typings/Control.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ export interface ControlProps {
     args: { value: string | number; checked: boolean },
     e: React.MouseEvent
   ) => void;
-  value: string | number;
+  value: number | string | {[key : string | number ]: stirng | number};
```
Comment (azizhk) [inline src/components/typings/Control.ts]: `stirng`
and also this almost makes this the `any` type.
I am open to making it a Generic Type
Like Controls<T>
Something like https://github.com/ritz078/pebble/blob/master/stories/optionGroupSection.story.tsx#L21

Comment (ritz078) [reply src/components/typings/Control.ts]: The one reason I was against object as value was that I can't determine by comparison which value is actually selected. For that I would have had to use a different prop like valueExtractor. That's just like making a simple thing complex.

Comment (azizhk) [reply src/components/typings/Control.ts]: Well, even right now an array of objects is working as we are using `Array.prototype.includes` to determine if a value is selected. So there is no need to add valueExtractor.

**Comment by azizhk** [comment]
Done better in https://github.com/ritz078/pebble/pull/191
Though don't know why the changes in Calendar were there.

**Pull request closed without merging**